### PR TITLE
media-gfx/cura-4.2.1: version bump

### PR DESCRIPTION
**Pull request by Perlovka**
Cura 4.2.1 and dependencies version bump

### CODE REVIEW

**Comment by Perlovka** [comment]
@Amynka FYI, PR is already here :wink: (https://github.com/gentoo/gentoo/pull/9199#issuecomment-405309130)

**Comment by juippis** [comment]
```
 * ERROR: dev-python/uranium-4.2.0 failed (test phase):
 *   emake failed
```
could you look into this?

File: dev-python/uranium/uranium-4.2.0.ebuild [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+	~dev-libs/libarcus-${PV}:=[python,${PYTHON_USEDEP}]
+	dev-python/PyQt5[${PYTHON_USEDEP},declarative,network,svg]
+	dev-python/numpy[${PYTHON_USEDEP}]
+	>=sci-libs/scipy-1.1[${PYTHON_USEDEP}]
+	sci-libs/Shapely[${PYTHON_USEDEP}]
+	dev-qt/qtquickcontrols:5
+	dev-qt/qtquickcontrols2:5"
+
+DEPEND="${RDEPEND}
+	sys-devel/gettext
```
Comment (juippis) [inline dev-python/uranium/uranium-4.2.0.ebuild]: gettext and doxygen most likely belong to BDEPEND with EAPI-7.

**Comment by juippis** [comment]
Seems like I can't even build curaengine, 
```
FAILED: CMakeFiles/_CuraEngine.dir/src/layerPart.cpp.o 
/usr/bin/x86_64-pc-linux-gnu-g++ -DARCUS -DBUILD_TESTS -DBUILD_TESTS=1 -DVERSION=\"master\" -I/usr/include/stb -I. -I/tmp/portage/media-gfx/curaengine-4.2.0/work/CuraEngine-4.2.0/libs -I/tmp/portage/media-gfx/curaengine-4.2.0/work/CuraEngine-4.2.0/libs/clipper  -DNDEBUG -O3 -DNDEBUG -Wall -static-libstdc++ -fopenmp   -std=gnu++11 -MD -MT CMakeFiles/_CuraEngine.dir/src/layerPart.cpp.o -MF CMakeFiles/_CuraEngine.dir/src/layerPart.cpp.o.d -o CMakeFiles/_CuraEngine.dir/src/layerPart.cpp.o -c /tmp/portage/media-gfx/curaengine-4.2.0/work/CuraEngine-4.2.0/src/layerPart.cpp
/tmp/portage/media-gfx/curaengine-4.2.0/work/CuraEngine-4.2.0/src/layerPart.cpp: In function 'void cura::createLayerParts(cura::SliceMeshStorage&, cura::Slicer*)':
/tmp/portage/media-gfx/curaengine-4.2.0/work/CuraEngine-4.2.0/src/layerPart.cpp:57:5: error: 'total_layers' not specified in enclosing 'parallel'
   57 |     for (int layer_nr = 0; layer_nr < static_cast<int>(total_layers); layer_nr++)
      |     ^~~
/tmp/portage/media-gfx/curaengine-4.2.0/work/CuraEngine-4.2.0/src/layerPart.cpp:55:9: error: enclosing 'parallel'
   55 | #pragma omp parallel for default(none) shared(mesh, slicer) schedule(dynamic)
      |         ^~~
ninja: build stopped: subcommand failed.
```
any ideas?

**Comment by MathyV** [comment]
@juippis are you using gcc-9? https://bugs.gentoo.org/685178

**Comment by MathyV** [comment]
I can confirm everything builds for me with gcc-8.3.0. Sadly some functionality is still broken running as a regular user as described here https://github.com/Ultimaker/Cura/issues/5488

**Comment by juippis** [comment]
Yes. You then have to fix the "broken" ebuild to depend on a working gcc, and make a check that such version of gcc is being used when trying to compile that package.

**Comment by MathyV** [comment]
Upstream apparently still is discussing the best way to tackle this, I think depending on gcc-8 is the best solution for now. @Perlovka can you update the PR?

**Comment by Perlovka** [comment]
@MathyV <=gcc-8 for all ebuilds? gcc-9 is masked on my machines, can't check right now.
I'll try to fix ebuilds tomorrow.

**Comment by juippis** [comment]
Remember that you need to use either `gcc-*version` checks from toolchain-funcs.eclass or `has_version` to make it work. There are multiple examples in the tree.

**Comment by MathyV** [comment]
I don't have gcc-9 either so I can't check but in general, would anyone use
two compilers to build the application and dependencies?

**Comment by Perlovka** [comment]
> could you look into this?

Well, I have no idea why src_test() is in ebuild, probably from ancient version.
I'll drop this.

File: dev-python/uranium/uranium-4.2.0.ebuild (hunk 1 above)
Comment (Perlovka) [reply dev-python/uranium/uranium-4.2.0.ebuild]: Fixed, also in cura and curaengine

**Comment by Perlovka** [comment]
Seems like only curaengine can't be built with gcc-9 or clang-8 (https://github.com/Ultimaker/CuraEngine/issues/984#issuecomment-499076643), so added checks for clang as well.

Fixed BDEPENDS where applicable.

**Comment by juippis** [comment]
3 packages failed to compile with FEATURES="test" USE="test". I expect you to fix or RESTRICT them. 

```
0% tests passed, 4 tests failed out of 4

Total Test time (real) =   0.33 sec

The following tests FAILED:
	  1 - pytest-main (Failed)
	  2 - pytest-OBJReader (Failed)
	  3 - pytest-STLReader (Failed)
	  4 - code-style (Failed)
Errors while running CTest
 * ERROR: dev-python/uranium-4.2.0::testworld failed (test phase):
```

```
Cloning into 'gmock'...
fatal: unable to access 'https://github.com/google/googletest.git/': Couldn't resolve host 'github.com'
Cloning into 'gmock'...
fatal: unable to access 'https://github.com/google/googletest.git/': Couldn't resolve host 'github.com'
Cloning into 'gmock'...
fatal: unable to access 'https://github.com/google/googletest.git/': Couldn't resolve host 'github.com'
CMake Error at /tmp/portage/media-gfx/curaengine-4.2.0/work/curaengine-4.2.0_build/gmock/tmp/gmock-gitclone.cmake:31 (message):
  Failed to clone repository: 'https://github.com/google/googletest.git'


-- Had to git clone more than once:
          3 times.

/tmp/portage/media-gfx/curaengine-4.2.0/work/CuraEngine-4.2.0/libs/clipper/clipper.cpp:66:8: note: ‘struct ClipperLib::TEdge’ declared here
 struct TEdge {
        ^~~~~
ninja: build stopped: subcommand failed.
 * ERROR: media-gfx/curaengine-4.2.0::testworld failed (compile phase):
```

```
10% tests passed, 9 tests failed out of 10

Total Test time (real) =   0.44 sec

The following tests FAILED:
	  1 - pytest-main (Failed)
	  2 - pytest-FirmwareUpdateChecker (Failed)
	  3 - pytest-LegacyProfileReader (Failed)
	  4 - pytest-UM3NetworkPrinting (Failed)
	  5 - pytest-VersionUpgrade25to26 (Failed)
	  6 - pytest-VersionUpgrade26to27 (Failed)
	  7 - pytest-VersionUpgrade27to30 (Failed)
	  8 - pytest-VersionUpgrade34to35 (Failed)
	  9 - code-style (Failed)
Errors while running CTest
 * ERROR: media-gfx/cura-4.2.1::testworld failed (test phase):
```


Then there's this: 
```
 * Messages for package media-gfx/curaengine-4.2.0:

 * QA Notice: Files built without respecting CFLAGS have been detected
 *  Please include the following list of files in your report:
 * /usr/bin/CuraEngine
```

I'll open bug reports for these errors later when I have more time, but for now, let's get this merged. Thanks!

**Pull request closed without merging**